File: forntend_quanlynhansu/src/services/UserService.js
import api from "./BaseApi";

const UserService = {
  getAll: () => api.get("/users"),
  getById: (id) => api.get(`/users/${id}`),
  create: (data) => api.post("/users", data),
  update: (id, formData) =>
    api.put(`/users/${id}`, formData, {
      headers: { "Content-Type": "multipart/form-data" },
    }),
  delete: (id) => api.delete(`/users/${id}`),
  uploadImage: (id, file) => {
    const formData = new FormData();
    formData.append("file", file);
    return api.post(`/users/${id}/image`, formData, {
      headers: { "Content-Type": "multipart/form-data" },
    });
  },
};

export default UserService;

File: src/main/java/com/example/quanlynhansu/service/UserService.java
package com.example.quanlynhansu.service;

import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.example.quanlynhansu.entity.User;

public interface UserService {
    User create(User user);
    User update(Long id, User user, MultipartFile file);
    void delete(Long id);
    User getById(Long id);
    List<User> getAll();
    String uploadImage(Long id, MultipartFile file);
    String uploadFaceImage(Long userId, MultipartFile file);
}
